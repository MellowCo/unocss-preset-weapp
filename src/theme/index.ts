/* @export-submodules */
export * from './animation'
export * from './colors'
export * from './default'
export * from './filters'
export * from './font'
export * from './misc'
export * from './preflight'
export * from './size'
export * from './types'
