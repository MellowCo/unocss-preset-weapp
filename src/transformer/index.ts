import { defaultAttributes, defaultIgnoreNonValuedAttributes } from 'unplugin-attributify-to-class/utils'
import { defaultRules } from 'unplugin-transform-class/utils'
import transformerClass from './transformerClass'
import { extractorAttributify, transformerAttributify } from './transformerAttributify'

export {
  transformerClass,
  extractorAttributify,
  transformerAttributify,
  defaultAttributes,
  defaultIgnoreNonValuedAttributes,
  defaultRules,
}
