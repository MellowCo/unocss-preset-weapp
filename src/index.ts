import type { Theme, ThemeAnimation } from './theme'
import { definePreset, type PresetOptions } from '@unocss/core'
import { extractorArbitraryVariants } from '@unocss/extractor-arbitrary-variants'
import { cacheTransformEscapESelector, defaultRules } from 'unplugin-transform-class/utils'
import preflights from './preflights'
import { taroCssPxTransform, taroH5CssRemTransform, uniAppVue2CssRpxTransform } from './rpxTranform'
import { rules } from './rules'
import { shorthands } from './shorthands'
import { theme } from './theme'

import { variants } from './variants'

export { colors, theme } from './theme'
export { parseColor } from './utils'

// support custom shadow color
export type { Theme, ThemeAnimation }

export interface DarkModeSelectors {
  /**
   * Selector for light variant.
   *
   * @default '.light'
   */
  light?: string

  /**
   * Selector for dark variant.
   *
   * @default '.dark'
   */
  dark?: string
}

export interface PresetWeappOptions extends PresetOptions {
  /**
   * Dark mode options
   *
   */
  dark?: 'class' | 'media' | DarkModeSelectors

  /**
   * @default false
   */
  attributifyPseudo?: boolean

  /**
   * Prefix for CSS variables.
   *
   * @default 'un-'
   */
  variablePrefix?: string

  /**
   * Utils prefix
   *
   * @default undefined
   */
  prefix?: string | string[]

  /**
   * Generate preflight
   *
   * @default true
   */
  preflight?: boolean

  /**
   * 是否转换微信class
   *
   * @default true
   */
  transform?: boolean

  /**
   * 平台
   * @default 'uniapp'
   */
  platform?: 'taro' | 'uniapp'

  /**
   * taro h5 rem 换算尺寸标准
   * @default 750
   * @link https://taro-docs.jd.com/taro/docs/size
   */
  designWidth?: number

  /**
   * taro 设计稿尺寸换算规则
   * @default '{ 640: 2.34 / 2, 750: 1, 828: 1.81 / 2}'
   * @link https://taro-docs.jd.com/taro/docs/size
   */
  deviceRatio?: Record<number, number>

  /**
   * taro webpack 版本
   * taro webpack4 和 webpack5 h5根字体大小，导致不同版本 rem 不一致
   * 见下面issues
   * @link https://github.com/NervJS/taro/issues/12361
   * @default webpack4
   */
  taroWebpack?: 'webpack4' | 'webpack5'

  /**
   * 是否为h5
   * @default false
   */
  isH5?: boolean

  /**
   * 自定义转换规则
   * @default https://github.com/MellowCo/unplugin-transform-class#options
   */
  transformRules?: Record<string, string>

  /**
   * wh 是否使用 rpx 为默认单位
   * @example
   * ```
   * whRpx: true
   * w-10 -> width: 10rpx
   * h-10 -> height: 10rpx
   * ```
   *
   * @example
   * ```
   * whRpx: false
   * w-10 -> width: 80rpx
   * h-10 -> height: 80rpx
   * ```
   * @default true
   */
  whRpx?: boolean

  /**
   * Enable arbitrary variants, for example `<div class="[&>*]:m-1 [&[open]]:p-2"></div>`.
   *
   * Disable this might slightly improve the performance.
   *
   * @default true
   */
  arbitraryVariants?: boolean
}

export const presetWeapp = definePreset((options: PresetWeappOptions = {}) => {
  options.dark = options.dark ?? 'class'
  options.attributifyPseudo = options.attributifyPseudo ?? false
  options.preflight = options.preflight ?? true
  options.variablePrefix = options.variablePrefix ?? 'un-'

  options.transform = options.transform ?? true
  options.isH5 = options.isH5 ?? false
  options.designWidth = options.designWidth ?? 750
  options.deviceRatio = options.deviceRatio ?? {
    640: 2.34 / 2,
    750: 1,
    828: 1.81 / 2,
  }
  options.platform = options.platform ?? 'uniapp'
  options.taroWebpack = options.taroWebpack ?? 'webpack4'
  options.transformRules = options.transformRules ?? defaultRules
  options.whRpx = options.whRpx ?? true

  return {
    name: 'unocss-preset-weapp',
    theme: {
      ...theme,
      transformRules: options.transformRules,
      whRpx: options.whRpx,
    },
    rules,
    variants: variants(options),
    options,
    postprocess(css) {
      // 是否转义class
      if (options.transform)
        css.selector = cacheTransformEscapESelector(css.selector, options.transformRules)

      // 设置变量前缀
      if (options.variablePrefix !== 'un-')
        VarPrefixPostprocessor(options.variablePrefix!, css)

      // taro 处理 h5 和 小程序 px 和 rpx 转换
      if (options.platform === 'taro') {
        const { taroWebpack, designWidth, deviceRatio } = options

        if (options.isH5) {
          // h5 px rpx 转 rem
          taroH5CssRemTransform(css, taroWebpack!, designWidth!, deviceRatio!)
        }
        else {
          // 小程序 taro 处理 px 为 rpx
          taroCssPxTransform(css, designWidth!, deviceRatio!)
        }
      }

      // uniapp vue2 webpack: h5 rpx 处理
      if (options.platform === 'uniapp' && options.isH5)
        uniAppVue2CssRpxTransform(css)
    },
    preflights: options.preflight ? preflights(options.isH5, options.platform) : [],
    prefix: options.prefix,
    extractorDefault: options.arbitraryVariants === false
      ? undefined
      : extractorArbitraryVariants(),
    autocomplete: {
      shorthands,
    },
  }
})

function VarPrefixPostprocessor(prefix: string, obj: any) {
  obj.entries.forEach((i: any) => {
    i[0] = i[0].replace(/^--un-/, `--${prefix}`)
    if (typeof i[1] === 'string')
      i[1] = i[1].replace(/var\(--un-/g, `var(--${prefix}`)
  })
}

export default presetWeapp
