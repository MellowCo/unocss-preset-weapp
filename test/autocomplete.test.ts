import { createAutocomplete, parseAutocomplete } from '@unocss/autocomplete'
import { createGenerator } from '@unocss/core'
import { describe, expect, it } from 'vitest'
import presetWeapp from '../src/index'
import { extractorAttributify } from '../src/transformer'

describe('autocomplete', () => {
  const { presetWeappAttributify } = extractorAttributify()

  const uno = createGenerator({
    presets: [
      presetWeapp(),
      presetWeappAttributify(),
    ],
    shortcuts: [
      {
        'foo': 'text-red',
        'foo-bar': 'text-red',
      },
      [/^bg-mode-(.+)$/, ([, mode]) => `bg-blend-${mode}`, { autocomplete: ['bg-mode-(color|normal)'] }],
    ],
  })

  const ac = createAutocomplete(uno)

  async function enumerateSuggestions(inputs: string[]) {
    return Object.fromEntries(await Promise.all(inputs.map(async input => [
      input,
      (await ac.suggest(input)).slice(0, 10).join(' '),
    ])))
  }

  const fixture = `
<div bg="blue-500">
  <div border="~ b-
</div>
`

  it('should resolve autocomplete config', () => {
    expect(ac.templates.length).toBeGreaterThan(0)

    ac.templates.forEach((i) => {
      if (typeof i === 'string')
        parseAutocomplete(i, uno.config.theme, uno.config.autocomplete.shorthands)
    })
  })

  it('should work', async () => {
    expect((await ac.suggest('m-1'))[0])
      .toMatchInlineSnapshot('"m-1"')

    expect((await ac.suggest('invalid'))[0])
      .not
      .toBe('invalid')
  })

  it('should provide autocomplete', async () => {
    expect(
      await enumerateSuggestions([
        // sort this list in alphabetical order
        'align-',
        'aspect-',
        'bg-',
        'bg-gradient-',
        'bg-r',
        'border',
        'border-',
        'border-r',
        'border-spacing-',
        'columns-',
        'divide-',
        'fill-',
        'filter-',
        'fle',
        'font-',
        'font-',
        'grow-',
        'keyframes-',
        'leading-',
        'line-clamp-',
        'm-',
        'max-w-',
        'mx-',
        'object-',
        'origin-',
        'outline-',
        'outline-offset-',
        'placeholder-',
        'scroll-',
        'scroll-m-',
        'shadow-',
        'space-',
        'text-r',
        'text-red-',
        'touch-',
        'transition-',
        'translate-',
        'translate-x-',
        'scale-',
        'scale-x-',
        'skew-x-',
        'transform-scale-z-',
        'transform-scale-x-',
        'transform-scale-',
        'transform-skew-x-',
        'transform-skew-y-',
        'grid-row',
        'grid-row-start-',
        'grid-row-end-',
        'grid-rows-',
        'grid-auto-flow-',
        'grid-flow-',
        'auto-flow-',
        'grid-auto-rows-',
        'v-',
        'w-',
        'z-',
      ]),
    ).toMatchSnapshot()
  })

  it('should not suggest blocked rules', async () => {
    const uno = createGenerator({
      presets: [
        presetWeapp(),
      ],
      blocklist: [
        /[A-Z]/,
      ],
    })

    const ac = createAutocomplete(uno)

    expect((await ac.suggest('text-trueGray-')))
      .toMatchInlineSnapshot('[]')
  })

  it('should provide skip DEFAULT', async () => {
    expect((await ac.suggest('text-red-')))
      .toMatchSnapshot()
  })

  it('should provide variants', async () => {
    expect(await ac.suggest('lt-'))
      .toMatchInlineSnapshot(`
        [
          "lt-lg",
          "lt-md",
          "lt-sm",
          "lt-xl",
          "ltr:",
          "lt-2xl",
        ]
      `)
  })

  it('should accept variants', async () => {
    expect(await ac.suggest('dark:md:m-'))
      .toMatchSnapshot()
  })

  it('should skip single-pass variants', async () => {
    expect(await ac.suggest('dark:dar')).not.toContain('dark:')
    expect(await ac.suggest('active:fir')).toContain('active:first:')
  })

  it('should support extractors', async () => {
    const res = await ac.suggestInFile(fixture, 40)

    expect(res.suggestions.every(i => i[0].startsWith('border-'))).toBeTruthy()
    expect(res.suggestions.some(i => i[1].startsWith('border-'))).toBeFalsy()

    const replacement = res.resolveReplacement(res.suggestions[0][0])
    expect(replacement).toMatchInlineSnapshot(`
      {
        "end": 40,
        "replacement": "b-amber",
        "start": 38,
      }
    `)

    expect(fixture.slice(0, replacement.start) + replacement.replacement + fixture.slice(replacement.end))
      .toMatchInlineSnapshot(`
        "
        <div bg="blue-500">
          <div border="~ b-amber
        </div>
        "
      `)
  })

  it('should escape regex', async () => {
    expect((await ac.suggest('m-['))[0])
      .toMatchInlineSnapshot('undefined')
  })

  it('should suggest static shortcuts', async () => {
    expect(await ac.suggest('foo'))
      .toMatchInlineSnapshot(`
        [
          "foo",
          "foo-bar",
        ]
      `)
  })

  it('should suggest shortcuts with autocomplete key', async () => {
    expect(await ac.suggest('bg-mode-'))
      .toMatchInlineSnapshot(`
        [
          "bg-mode-color",
          "bg-mode-normal",
        ]
      `)
  })
})

describe('use uno cache', () => {
  const uno = createGenerator({
    presets: [
      presetWeapp(),
    ],
    shortcuts: [
      {
        'foo': 'text-red',
        'foo-bar': 'text-red',
      },
      [/^btn-(red|green)$/, m => `text-${m[1]}`],
    ],
  })

  const ac = createAutocomplete(uno)

  it('use cache', async () => {
    expect(await ac.suggest('btn'))
      .toMatchInlineSnapshot(`
        [
          "b-t-neutral",
          "b-t-none",
        ]
      `)
    await uno.generate('btn-red btn-green m-100', { preflights: false })
    ac.reset()

    expect(await ac.suggest('btn'))
      .toMatchInlineSnapshot(`
        [
          "b-t-neutral",
          "b-t-none",
          "btn-green",
          "btn-red",
        ]
      `)
    expect(await ac.suggest('m-1'))
      .toMatchInlineSnapshot(`
        [
          "m-1",
          "m-100",
        ]
      `)
  })
})
