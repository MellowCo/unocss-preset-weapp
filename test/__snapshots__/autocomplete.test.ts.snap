// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`autocomplete > should accept variants 1`] = `
[
  "dark:md:m-xy",
  "dark:md:marker:",
  "dark:md:max-block",
  "dark:md:max-block-auto",
  "dark:md:max-block-full",
  "dark:md:max-block-half",
  "dark:md:max-block-lg",
  "dark:md:max-block-md",
  "dark:md:max-block-none",
  "dark:md:max-block-screen",
  "dark:md:max-block-sm",
  "dark:md:max-block-xl",
  "dark:md:max-block-xs",
  "dark:md:max-h",
  "dark:md:max-h-auto",
  "dark:md:max-h-full",
  "dark:md:max-h-half",
  "dark:md:max-h-lg",
  "dark:md:max-h-md",
  "dark:md:max-h-none",
  "dark:md:max-h-screen",
  "dark:md:max-h-screen-lg",
  "dark:md:max-h-screen-md",
  "dark:md:max-h-screen-sm",
  "dark:md:max-h-screen-xl",
  "dark:md:max-h-sm",
  "dark:md:max-h-xl",
  "dark:md:max-h-xs",
  "dark:md:max-inline",
  "dark:md:max-inline-auto",
  "dark:md:max-inline-full",
  "dark:md:max-inline-half",
  "dark:md:max-inline-lg",
  "dark:md:max-inline-md",
  "dark:md:max-inline-none",
  "dark:md:max-inline-screen",
  "dark:md:max-inline-sm",
  "dark:md:max-inline-xl",
  "dark:md:max-inline-xs",
  "dark:md:max-lg",
  "dark:md:max-md",
  "dark:md:max-sm",
  "dark:md:max-w",
  "dark:md:max-w-auto",
  "dark:md:max-w-full",
  "dark:md:max-w-half",
  "dark:md:max-w-lg",
  "dark:md:max-w-md",
  "dark:md:max-w-none",
  "dark:md:max-w-screen",
  "dark:md:max-w-screen-lg",
  "dark:md:max-w-screen-md",
  "dark:md:max-w-screen-sm",
  "dark:md:max-w-screen-xl",
  "dark:md:max-w-sm",
  "dark:md:max-w-xl",
  "dark:md:max-w-xs",
  "dark:md:max-xl",
  "dark:md:min-block",
  "dark:md:min-block-auto",
  "dark:md:min-block-full",
  "dark:md:min-block-half",
  "dark:md:min-block-lg",
  "dark:md:min-block-md",
  "dark:md:min-block-none",
  "dark:md:min-block-screen",
  "dark:md:min-block-sm",
  "dark:md:min-block-xl",
  "dark:md:min-block-xs",
  "dark:md:min-h",
  "dark:md:min-h-auto",
  "dark:md:min-h-full",
  "dark:md:min-h-half",
  "dark:md:min-h-lg",
  "dark:md:min-h-md",
  "dark:md:min-h-none",
  "dark:md:min-h-screen",
  "dark:md:min-h-screen-lg",
  "dark:md:min-h-screen-md",
  "dark:md:min-h-screen-sm",
  "dark:md:min-h-screen-xl",
  "dark:md:min-h-sm",
  "dark:md:min-h-xl",
  "dark:md:min-h-xs",
  "dark:md:min-inline",
  "dark:md:min-inline-auto",
  "dark:md:min-inline-full",
  "dark:md:min-inline-half",
  "dark:md:min-inline-lg",
  "dark:md:min-inline-md",
  "dark:md:min-inline-none",
  "dark:md:min-inline-screen",
  "dark:md:min-inline-sm",
  "dark:md:min-inline-xl",
  "dark:md:min-inline-xs",
  "dark:md:min-w",
  "dark:md:min-w-auto",
  "dark:md:min-w-full",
  "dark:md:min-w-half",
  "dark:md:min-w-lg",
  "dark:md:min-w-md",
  "dark:md:min-w-none",
  "dark:md:min-w-screen",
  "dark:md:min-w-screen-lg",
  "dark:md:min-w-screen-md",
  "dark:md:min-w-screen-sm",
  "dark:md:min-w-screen-xl",
  "dark:md:min-w-sm",
  "dark:md:min-w-xl",
  "dark:md:min-w-xs",
  "dark:md:m-bs-0",
  "dark:md:m-be-0",
  "dark:md:m-is-0",
  "dark:md:m-ie-0",
  "dark:md:m-bs-1",
  "dark:md:m-be-1",
  "dark:md:m-is-1",
  "dark:md:m-ie-1",
  "dark:md:m-bs-2",
  "dark:md:m-be-2",
  "dark:md:m-is-2",
  "dark:md:m-ie-2",
  "dark:md:m-bs-3",
  "dark:md:m-be-3",
  "dark:md:m-is-3",
  "dark:md:m-ie-3",
  "dark:md:m-bs-4",
  "dark:md:m-be-4",
  "dark:md:m-is-4",
  "dark:md:m-ie-4",
  "dark:md:m-bs-5",
  "dark:md:m-be-5",
  "dark:md:m-is-5",
  "dark:md:m-ie-5",
  "dark:md:m-bs-6",
  "dark:md:m-be-6",
  "dark:md:m-is-6",
  "dark:md:m-ie-6",
  "dark:md:m-bs-8",
  "dark:md:m-be-8",
  "dark:md:m-is-8",
  "dark:md:m-ie-8",
  "dark:md:m-bs-10",
  "dark:md:m-be-10",
  "dark:md:m-is-10",
  "dark:md:m-ie-10",
  "dark:md:m-bs-12",
  "dark:md:m-be-12",
  "dark:md:m-is-12",
  "dark:md:m-ie-12",
  "dark:md:m-bs-24",
  "dark:md:m-be-24",
  "dark:md:m-is-24",
  "dark:md:m-ie-24",
  "dark:md:m-bs-36",
  "dark:md:m-be-36",
  "dark:md:m-is-36",
  "dark:md:m-ie-36",
  "dark:md:m-block-0",
  "dark:md:m-inline-0",
  "dark:md:m-block-1",
  "dark:md:m-inline-1",
  "dark:md:m-block-2",
  "dark:md:m-inline-2",
  "dark:md:m-block-3",
  "dark:md:m-inline-3",
  "dark:md:m-block-4",
  "dark:md:m-inline-4",
  "dark:md:m-block-5",
  "dark:md:m-inline-5",
  "dark:md:m-block-6",
  "dark:md:m-inline-6",
  "dark:md:m-block-8",
  "dark:md:m-inline-8",
  "dark:md:m-block-10",
  "dark:md:m-inline-10",
  "dark:md:m-block-12",
  "dark:md:m-inline-12",
  "dark:md:m-block-24",
  "dark:md:m-inline-24",
  "dark:md:m-block-36",
  "dark:md:m-inline-36",
  "dark:md:mx-0",
  "dark:md:my-0",
  "dark:md:mt-0",
  "dark:md:mb-0",
  "dark:md:ml-0",
  "dark:md:mr-0",
  "dark:md:ms-0",
  "dark:md:me-0",
  "dark:md:mx-1",
  "dark:md:my-1",
  "dark:md:mt-1",
  "dark:md:mb-1",
  "dark:md:ml-1",
  "dark:md:mr-1",
  "dark:md:ms-1",
  "dark:md:me-1",
  "dark:md:mx-2",
  "dark:md:my-2",
  "dark:md:mt-2",
  "dark:md:mb-2",
  "dark:md:ml-2",
  "dark:md:mr-2",
  "dark:md:ms-2",
  "dark:md:me-2",
  "dark:md:mx-3",
  "dark:md:my-3",
  "dark:md:mt-3",
  "dark:md:mb-3",
  "dark:md:ml-3",
  "dark:md:mr-3",
  "dark:md:ms-3",
  "dark:md:me-3",
  "dark:md:mx-4",
  "dark:md:my-4",
  "dark:md:mt-4",
  "dark:md:mb-4",
  "dark:md:ml-4",
  "dark:md:mr-4",
  "dark:md:ms-4",
  "dark:md:me-4",
  "dark:md:mx-5",
  "dark:md:my-5",
  "dark:md:mt-5",
  "dark:md:mb-5",
  "dark:md:ml-5",
  "dark:md:mr-5",
  "dark:md:ms-5",
  "dark:md:me-5",
  "dark:md:mx-6",
  "dark:md:my-6",
  "dark:md:mt-6",
  "dark:md:mb-6",
  "dark:md:ml-6",
  "dark:md:mr-6",
  "dark:md:ms-6",
  "dark:md:me-6",
  "dark:md:mx-8",
  "dark:md:my-8",
  "dark:md:mt-8",
  "dark:md:mb-8",
  "dark:md:ml-8",
  "dark:md:mr-8",
  "dark:md:ms-8",
  "dark:md:me-8",
  "dark:md:mx-10",
  "dark:md:my-10",
  "dark:md:mt-10",
  "dark:md:mb-10",
  "dark:md:ml-10",
  "dark:md:mr-10",
  "dark:md:ms-10",
  "dark:md:me-10",
  "dark:md:mx-12",
  "dark:md:my-12",
  "dark:md:mt-12",
  "dark:md:mb-12",
  "dark:md:ml-12",
  "dark:md:mr-12",
  "dark:md:ms-12",
  "dark:md:me-12",
  "dark:md:mx-24",
  "dark:md:my-24",
  "dark:md:mt-24",
  "dark:md:mb-24",
  "dark:md:ml-24",
  "dark:md:mr-24",
  "dark:md:ms-24",
  "dark:md:me-24",
  "dark:md:mx-36",
  "dark:md:my-36",
  "dark:md:mt-36",
  "dark:md:mb-36",
  "dark:md:ml-36",
  "dark:md:mr-36",
  "dark:md:ms-36",
  "dark:md:me-36",
  "dark:md:m0",
  "dark:md:m1",
  "dark:md:m2",
  "dark:md:m3",
  "dark:md:m4",
  "dark:md:m5",
  "dark:md:m6",
  "dark:md:m8",
  "dark:md:m10",
  "dark:md:m12",
  "dark:md:m24",
  "dark:md:m36",
  "dark:md:m-0",
  "dark:md:m-1",
  "dark:md:m-2",
  "dark:md:m-3",
  "dark:md:m-4",
  "dark:md:m-5",
  "dark:md:m-6",
  "dark:md:m-8",
  "dark:md:m-10",
  "dark:md:m-12",
  "dark:md:m-24",
  "dark:md:m-36",
  "dark:md:min-h-screen-2xl",
  "dark:md:max-h-screen-2xl",
  "dark:md:min-w-screen-2xl",
  "dark:md:max-w-screen-2xl",
  "dark:md:max-w-2xl",
  "dark:md:max-h-2xl",
  "dark:md:max-block-2xl",
  "dark:md:max-inline-2xl",
  "dark:md:min-w-2xl",
  "dark:md:min-h-2xl",
  "dark:md:min-block-2xl",
  "dark:md:min-inline-2xl",
  "dark:md:max-w-3xl",
  "dark:md:max-h-3xl",
  "dark:md:max-block-3xl",
  "dark:md:max-inline-3xl",
  "dark:md:min-w-3xl",
  "dark:md:min-h-3xl",
  "dark:md:min-block-3xl",
  "dark:md:min-inline-3xl",
  "dark:md:max-w-4xl",
  "dark:md:max-h-4xl",
  "dark:md:max-block-4xl",
  "dark:md:max-inline-4xl",
  "dark:md:min-w-4xl",
  "dark:md:min-h-4xl",
  "dark:md:min-block-4xl",
  "dark:md:min-inline-4xl",
  "dark:md:max-w-5xl",
  "dark:md:max-h-5xl",
  "dark:md:max-block-5xl",
  "dark:md:max-inline-5xl",
  "dark:md:min-w-5xl",
  "dark:md:min-h-5xl",
  "dark:md:min-block-5xl",
  "dark:md:min-inline-5xl",
  "dark:md:max-w-6xl",
  "dark:md:max-h-6xl",
  "dark:md:max-block-6xl",
  "dark:md:max-inline-6xl",
  "dark:md:min-w-6xl",
  "dark:md:min-h-6xl",
  "dark:md:min-block-6xl",
  "dark:md:min-inline-6xl",
  "dark:md:max-w-7xl",
  "dark:md:max-h-7xl",
  "dark:md:max-block-7xl",
  "dark:md:max-inline-7xl",
  "dark:md:min-w-7xl",
  "dark:md:min-h-7xl",
  "dark:md:min-block-7xl",
  "dark:md:min-inline-7xl",
  "dark:md:max-w-8xl",
  "dark:md:max-h-8xl",
  "dark:md:max-block-8xl",
  "dark:md:max-inline-8xl",
  "dark:md:min-w-8xl",
  "dark:md:min-h-8xl",
  "dark:md:min-block-8xl",
  "dark:md:min-inline-8xl",
  "dark:md:max-w-9xl",
  "dark:md:max-h-9xl",
  "dark:md:max-block-9xl",
  "dark:md:max-inline-9xl",
  "dark:md:min-w-9xl",
  "dark:md:min-h-9xl",
  "dark:md:min-block-9xl",
  "dark:md:min-inline-9xl",
  "dark:md:max-2xl",
]
`;

exports[`autocomplete > should provide autocomplete 1`] = `
{
  "align-": "align-base align-baseline align-bottom align-btm align-end align-inherit align-initial align-mid align-middle align-revert",
  "aspect-": "aspect-ratio aspect-ratio-square aspect-ratio-video aspect-square aspect-video",
  "auto-flow-": "",
  "bg-": "b-gray b-green b-groove bg-amber bg-auto bg-black bg-blend-color bg-blend-color-burn bg-blend-color-dodge bg-blend-darken",
  "bg-gradient-": "bg-gradient-conic bg-gradient-from bg-gradient-from-amber bg-gradient-from-black bg-gradient-from-blue bg-gradient-from-bluegray bg-gradient-from-blueGray bg-gradient-from-coolgray bg-gradient-from-coolGray bg-gradient-from-current",
  "bg-r": "b-gray b-green b-groove bg-red bg-repeat bg-repeat-inherit bg-repeat-initial bg-repeat-revert bg-repeat-revert-layer bg-repeat-round",
  "border": "border border-amber border-b border-b-amber border-b-black border-b-blue border-b-bluegray border-b-blueGray border-b-coolgray border-b-coolGray",
  "border-": "border-amber border-b border-b-amber border-b-black border-b-blue border-b-bluegray border-b-blueGray border-b-coolgray border-b-coolGray border-b-current",
  "border-r": "border-r border-r-amber border-r-black border-r-blue border-r-bluegray border-r-blueGray border-r-coolgray border-r-coolGray border-r-current border-r-cyan",
  "border-spacing-": "",
  "columns-": "",
  "divide-": "divide-amber divide-black divide-block divide-block-none divide-block-reverse divide-blue divide-bluegray divide-blueGray divide-coolgray divide-coolGray",
  "fill-": "fill-amber fill-black fill-blue fill-bluegray fill-blueGray fill-coolgray fill-coolGray fill-current fill-cyan fill-dark",
  "filter-": "filter-blur filter-blur-lg filter-blur-md filter-blur-sm filter-blur-xl filter-drop filter-drop-shadow filter-drop-shadow-color filter-drop-shadow-color-amber filter-drop-shadow-color-black",
  "fle": "flex flex-auto flex-basis-lg flex-basis-md flex-basis-none flex-basis-sm flex-basis-xl flex-basis-xs flex-col flex-col-reverse",
  "font-": "font-black font-bold font-extrabold font-extralight font-italic font-light font-medium font-mono font-normal font-not-italic",
  "grid-auto-flow-": "",
  "grid-auto-rows-": "",
  "grid-flow-": "",
  "grid-row": "grid-rows-none grid-rows-subgrid",
  "grid-row-end-": "",
  "grid-row-start-": "",
  "grid-rows-": "grid-rows-none grid-rows-subgrid",
  "grow-": "grow-0 grow-1 grow-2 grow-3 grow-4 grow-5 grow-6 grow-8 grow-10 grow-12",
  "keyframes-": "keyframes-back-in-down keyframes-back-in-left keyframes-back-in-right keyframes-back-in-up keyframes-back-out-down keyframes-back-out-left keyframes-back-out-right keyframes-back-out-up keyframes-bounce keyframes-bounce-alt",
  "leading-": "leading-loose leading-none leading-normal leading-relaxed leading-snug leading-tight",
  "line-clamp-": "line-clamp-inherit line-clamp-initial line-clamp-none line-clamp-revert line-clamp-revert-layer line-clamp-unset line-clamp-0 line-clamp-1 line-clamp-2 line-clamp-3",
  "m-": "m-xy marker: max-block max-block-auto max-block-full max-block-half max-block-lg max-block-md max-block-none max-block-screen",
  "max-w-": "max-w-auto max-w-full max-w-half max-w-lg max-w-md max-w-none max-w-screen max-w-sm max-w-xl max-w-xs",
  "mx-": "m-xy mx-0 mx-1 mx-2 mx-3 mx-4 mx-5 mx-6 mx-8 mx-10",
  "object-": "object-b object-bc object-bl object-bottom object-bottom-center object-bottom-left object-bottom-right object-br object-c object-cb",
  "origin-": "origin-b origin-bc origin-bl origin-bottom origin-bottom-center origin-bottom-left origin-bottom-right origin-br origin-c origin-cb",
  "outline-": "outline-amber outline-auto outline-black outline-blue outline-bluegray outline-blueGray outline-coolgray outline-coolGray outline-current outline-cyan",
  "outline-offset-": "outline-offset-0 outline-offset-1 outline-offset-2 outline-offset-3 outline-offset-4 outline-offset-5 outline-offset-6 outline-offset-8 outline-offset-10 outline-offset-12",
  "placeholder-": "placeholder-*: placeholder-active: placeholder-after: placeholder-align-base placeholder-align-baseline placeholder-align-bottom placeholder-align-btm placeholder-align-end placeholder-align-inherit placeholder-align-initial",
  "scale-": "scale-0 scale-10 scale-20 scale-30 scale-40 scale-50 scale-60 scale-70 scale-80 scale-90",
  "scale-x-": "scale-x-0 scale-x-10 scale-x-20 scale-x-30 scale-x-40 scale-x-50 scale-x-60 scale-x-70 scale-x-80 scale-x-90",
  "scroll-": "",
  "scroll-m-": "",
  "shadow-": "shadow-amber shadow-black shadow-blue shadow-bluegray shadow-blueGray shadow-coolgray shadow-coolGray shadow-current shadow-cyan shadow-dark",
  "skew-x-": "skew-x-0 skew-x-10 skew-x-20 skew-x-30 skew-x-40 skew-x-50 skew-x-60 skew-x-70 skew-x-80 skew-x-90",
  "space-": "space-block space-block-lg space-block-md space-block-none space-block-reverse space-block-sm space-block-xl space-block-xs space-inline space-inline-lg",
  "text-r": "text-red text-revert text-revert-layer text-right text-rose",
  "text-red-": "text-red-1 text-red-2 text-red-3 text-red-4 text-red-5 text-red-6 text-red-7 text-red-8 text-red-9 text-red-50",
  "touch-": "",
  "transform-scale-": "transform-scale-0 transform-scale-10 transform-scale-20 transform-scale-30 transform-scale-40 transform-scale-50 transform-scale-60 transform-scale-70 transform-scale-80 transform-scale-90",
  "transform-scale-x-": "transform-scale-x-0 transform-scale-x-10 transform-scale-x-20 transform-scale-x-30 transform-scale-x-40 transform-scale-x-50 transform-scale-x-60 transform-scale-x-70 transform-scale-x-80 transform-scale-x-90",
  "transform-scale-z-": "transform-scale-z-0 transform-scale-z-10 transform-scale-z-20 transform-scale-z-30 transform-scale-z-40 transform-scale-z-50 transform-scale-z-60 transform-scale-z-70 transform-scale-z-80 transform-scale-z-90",
  "transform-skew-x-": "transform-skew-x-0 transform-skew-x-10 transform-skew-x-20 transform-skew-x-30 transform-skew-x-40 transform-skew-x-50 transform-skew-x-60 transform-skew-x-70 transform-skew-x-80 transform-skew-x-90",
  "transform-skew-y-": "transform-skew-y-0 transform-skew-y-10 transform-skew-y-20 transform-skew-y-30 transform-skew-y-40 transform-skew-y-50 transform-skew-y-60 transform-skew-y-70 transform-skew-y-80 transform-skew-y-90",
  "transition-": "transition-all transition-colors transition-delay-none transition-discrete transition-duration-none transition-ease-DEFAULT transition-ease-in transition-ease-in-out transition-ease-linear transition-ease-out",
  "translate-": "translate-0 translate-10 translate-20 translate-30 translate-40 translate-50 translate-60 translate-70 translate-80 translate-90",
  "translate-x-": "translate-x-0 translate-x-10 translate-x-20 translate-x-30 translate-x-40 translate-x-50 translate-x-60 translate-x-70 translate-x-80 translate-x-90",
  "v-": "v-base v-baseline v-bottom v-btm v-end v-inherit v-initial v-mid v-middle v-revert",
  "w-": "w-auto w-full w-half w-lg w-md w-none w-screen w-screen-lg w-screen-md w-screen-sm",
  "z-": "z-0 z-1 z-2 z-3 z-4 z-5 z-6 z-8 z-10 z-12",
}
`;

exports[`autocomplete > should provide skip DEFAULT 1`] = `
[
  "text-red-1",
  "text-red-2",
  "text-red-3",
  "text-red-4",
  "text-red-5",
  "text-red-6",
  "text-red-7",
  "text-red-8",
  "text-red-9",
  "text-red-50",
  "text-red-100",
  "text-red-200",
  "text-red-300",
  "text-red-400",
  "text-red-500",
  "text-red-600",
  "text-red-700",
  "text-red-800",
  "text-red-900",
  "text-red-950",
]
`;
